### PR TITLE
feat: custom email auth

### PR DESCRIPTION
**Pull request by ice-orion**
* Remove Email-Password / Password-Reset flows
* Update firebase lib
* Add custom Email-Code flow

### CODE REVIEW

File: src/screens/AuthFlow/ConfirmEmailCode/index.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        </View>
+        <PrivacyTerms />
+      </View>
+      {validateLoading && <FullScreenLoading />}
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
```
Comment (ice-atlas) [inline src/screens/AuthFlow/ConfirmEmailCode/index.tsx]: use `commonStyles.flexOne`

**Review by ice-atlas — DISMISSED** [review]

**Review by ice-hades — DISMISSED** [review]

File: src/api/client/interceptors/request.ts [hunk 2]
```diff
@@ -24,6 +28,25 @@ async function onFulfilled(config: AxiosRequestConfig) {
     config.transformRequest = (data: FormData) => data;
   }
 
+  if (!config.headers?.['X-Language']) {
+    const appLocale = appLocaleSelector(store.getState());
```
Comment (ice-ares) [inline src/api/client/interceptors/request.ts:32]: This takes into account the user's language right?

File: src/api/client/interceptors/response.ts [hunk 3]
```diff
@@ -16,31 +21,76 @@ function onRejected(instance: AxiosInstance) {
...
+      tokenRefreshPromise = null;
+      store.dispatch(AccountActions.SET_TOKEN.STATE.create(newToken));
+    }
+
+    return newToken;
+  } catch (err) {
+    /**
+     * If the saved refresh token is not valid
+     */
+    if (is4xxApiError(err)) {
```
Comment (ice-ares) [inline src/api/client/interceptors/response.ts:89]: Check more stuff, like status code and maybe data in body

File: src/services/auth/index.ts [hunk 4]
```diff
@@ -159,12 +180,39 @@ export const onUserChanged = (listener: () => void) => {
...
+        accessToken: newFirebaseToken.token,
+        issuer: 'firebase',
+      } as const;
+    case 'custom':
+      const newCustomToken = await Api.auth.refreshTokens({
+        refreshToken: currentToken.refreshToken,
+      });
+      const token = {
+        ...newCustomToken,
+        issuer: 'custom',
```
Comment (ice-ares) [inline src/services/auth/index.ts:210]: @ice-cronus  Shouldnt this be ice or ice.io here?

File: src/screens/AuthFlow/ConfirmEmailCode/index.tsx (hunk 1 above)
Comment (ice-orion) [reply src/screens/AuthFlow/ConfirmEmailCode/index.tsx]: done

File: src/api/client/interceptors/request.ts (hunk 2 above)
Comment (ice-orion) [reply src/api/client/interceptors/request.ts:32]: Yes it takes user.language if authorized and device locale if not

File: src/services/auth/index.ts (hunk 4 above)
Comment (ice-cronus) [reply src/services/auth/index.ts:210]: It seems to be internal FE field to detect token type firebase or ice, not a issuer claim in the token. Probably it is better to rename it to provider as we do on BE, anyway it does not affect internal token's data where issuers are as follows:
```
	RefreshJwtIssuer            = "ice.io/refresh"
	AccessJwtIssuer             = "ice.io/access"
	MetadataIssuer              = "ice.io/metadata"
```

Comment (ice-orion) [reply src/services/auth/index.ts:210]: Yea that is an internal thing - either `firebase` or `custom` so we know how to process this token if we need to refresh it

File: src/api/client/interceptors/response.ts (hunk 3 above)
Comment (ice-orion) [reply src/api/client/interceptors/response.ts:89]: Decided to keep it as is

**Pull request merged**